component/azstorage/datalake.go

```diff
@@ -495,41 +495,52 @@ func (dl *Datalake) List(prefix string, marker *string, count int32) ([]*interna
 
 	// Process the paths returned in this result segment (if the segment is empty, the loop body won't execute)
 	for _, pathInfo := range listPath.Paths {
-		var mode fs.FileMode
-		if pathInfo.Permissions != nil {
-			mode, err = getFileMode(*pathInfo.Permissions)
+		var attr *internal.ObjAttr
+
+		if dl.Config.disableSymlink {
+			var mode fs.FileMode
```

Comment:
This peice shall be common. If disableSymlink is not there then we make an extra call to getAttr